src/scripts/modules/media/media.coffee

```diff
@@ -94,10 +94,23 @@ define (require) ->
       mainPage.regions.main.append(footerNav)
       footerNav.$el.addClass('footer-nav')
       @mainContent = windowWithSidebar.regions.main
-
       $pinnable = @regions.pinnable.$el
       pinnableTop = $pinnable.offset().top
       $toc = tocView.$el
+      
+      tocModel = tocView.model
+      @listenTo(tocModel, 'add:contents change:contents', ->
+        mediaBodyView.on('render', ->
+          $totalSubtitle = $("body").find("section [data-depth=1]")
+          subTable=$.map($totalSubtitle, (node) ->
+            subTitles=$(node).find('> [data-type="title"]').clone().text()
+            subIds="#" + $(node).attr("id")
+            return {text:subTitles,id:subIds}
+          )
+          tocModel.get('contents').models[1].attributes.contents.models[1].set({subTable: subTable})
```

Comment:
Why is the second item in `.models[]` updated instead of the first item? Can you add a JavaScript comment on this line so when someone else looks over this code in the future they will know what is happening?

Also, does this code cause an error when a person views a Page that is not in a book?